mlflow/sklearn/__init__.py

```diff
@@ -670,6 +676,10 @@ def fit_mlflow(self, func_name, *args, **kwargs):
 
         try_mlflow_log(log_model, self, artifact_path="model")
 
+        # If the estimator is of classifier type, then log corresponding metrics
+        if self._estimator_type == _CLASSIFIER:
```

Comment:
According to the sklearn docs on this subject (https://scikit-learn.org/stable/developers/develop.html#estimator-types), we should use utility methods, such as `sklearn.base.is_classifier`, to identify different types of models when possible:

```
... When a meta-estimator needs to distinguish among estimator types, instead of checking _estimator_type directly, helpers like base.is_classifier should be used.
```

Under the hood, these methods do seem to check the `_estimator_type` attribute, but that might change. In the case of clustering models, which don't have a nice `sklearn.base.*` method, we can still check for this attribute.

Finally, we should handle the case where `_estimator_type` is not defined by using `getattr` instead (see https://github.com/scikit-learn/scikit-learn/blob/2fa4272b1deee0ac605e173445659fc5fec293ef/sklearn/base.py#L777).